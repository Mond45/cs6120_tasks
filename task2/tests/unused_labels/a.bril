@no_label {
    a: int = const 0;
    b: int = const 1;
    a: int = add a b;
    print a;
}

@all_unused {
.label0:
    a: int = const 0;
.label1:
    b: int = const 1;
    a: int = add a b;
.label0:
    print a;
}

@some_used {
.unused:
    i: int = const 0;
    n: int = const 5;
.while_start:
    cond: bool = lt i n;
    br cond .while_body .while_end;
.while_body:
    nop;
    jmp .while_start;
.while_end:
    nop;
}

@all_used: int {
    call @no_label;
    i: int = const 0;
    n: int = const 5;
.while_start:
    cond: bool = lt i n;
    br cond .while_body .while_end;
.while_body:
    nop;
    jmp .while_start;
.while_end:
    ret i;
}
