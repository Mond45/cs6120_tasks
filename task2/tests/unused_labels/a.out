no_label: 
all_unused: label0, label1
some_used: unused
all_used: 
